frame/vault/src/lib.rs

```diff
@@ -487,6 +490,26 @@ pub mod pallet {
 			Self::deposit_event(Event::VaultStarted { vault });
 			Ok(().into())
 		}
+		#[pallet::weight(10_000)]
+		pub fn liquidate_strategy(
+			origin: OriginFor<T>,
+			vault_idx: VaultIndex,
+			strategy_account: T::AccountId,
+		) -> DispatchResult {
+			let from = ensure_signed(origin)?;
+			ensure!(Self::vault_info(&vault_idx)?.manager == from, Error::<T>::AccountIsNotManager);
+			let balance = CapitalStructure::<T>::try_get(&vault_idx, &strategy_account)
+				.map_err(|_err| DispatchError::CannotLookup)?
+				.balance;
+			<Self as StrategicVault>::deposit(&vault_idx, &strategy_account, balance)?;
+			Allocations::<T>::remove(&vault_idx, &strategy_account);
+			CapitalStructure::<T>::remove(&vault_idx, &strategy_account);
```

Comment:
The reasoning here is that if a strategy account withdrawn any amount of the vault's balance, then the same account should "repay" the vault before removal

Reply:
I think in this case we don't want to force the strategy to repay right now, as the strategy will call `available_funds` later and notice that it has been liquidated thanks to `FundsAvailability::MustLiquidate`. In this case, the strategy has to repay ASAP, but the funds might be elsewhere and take time to be rapatriated?

Reply:
so in this case liquidate must note remove, but set liquidation state to trigger Must Liquidate?

Reply:
According to `available_funds`, `MustLiquidate` will only be triggered if a vault does not exist or an `Allocation` does not exist or an existent vault is marked as stopped. That said, the only available option in the context of this PR is to remove the account `Allocation` but there are some caveats:

* Once removed, there is no way to re-created `Allocation` (within the crate) even if the account repays.
* `Allocation` is tied to `CapitalStructure` so it does not make sense, IMO, to have one without the other.

Reply:
There has been a patch in develop to merged `Allocation` with `CapitalStructure`. Sadly, we were using two branch until we decided to use the feature gate instead. We didn't merged the branches yet so this will result in a conflict while we will merge. In the context of this PR, removing both entry is correct

Reply:
OK, the call is now removing both storages but it is worth noting that `<Vaults as StrategicVault>::deposit` only works because of `ValueQuery`.